contracts/src/compute_engines/rebase_denomination/RebaseDenomination.cairo

```diff
@@ -0,0 +1,227 @@
+    let (
+        base_value, base_decimals, base_last_updated_timestamp, base_num_sources_aggregated
+    ) = IOracleController.get_value(
+        oracle_controller_address, base_asset_key, EmpiricAggregationModes.MEDIAN
+    )
+    if base_last_updated_timestamp == 0:
+        return (0, 0, 0, 0)
+    end
+
+    let (rebased_value, rebased_decimals) = _decimal_div(
+        quote_value, quote_decimals, base_value, base_decimals
+    )
+    let (last_updated_timestamp) = _max(quote_last_updated_timestamp, base_last_updated_timestamp)
+    let (num_sources_aggregated) = _max(quote_num_sources_aggregated, base_num_sources_aggregated)
+    return (rebased_value, rebased_decimals, last_updated_timestamp, num_sources_aggregated)
+end
+
+# @notice get address for admin
+# @return admin_address: returns admin's address
+@view
+func get_admin_address{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (
+    admin_address : felt
+):
+    let (admin_address) = Admin.get_admin_address()
+    return (admin_address)
+end
+
+# @notice get oracle controller for admin
+# @return oracle_controller_address: address for oracle controller
+@view
+func get_oracle_controller_address{
+    syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+}() -> (oracle_controller_address : felt):
+    let (oracle_controller_address) = oracle_controller_address_storage.read()
+    return (oracle_controller_address)
+end
+
+#
+# Setters
+#
+
+# @notice update admin address
+# @dev only the admin can set the new address
+# @param new_address
+@external
+func set_admin_address{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    new_address : felt
+):
+    Admin.only_admin()
+    Admin.set_admin_address(new_address)
+    return ()
+end
+
+# @notice update oracle controller address
+# @dev only the admin can update this
+# @param oracle_controller_address: new oracle controller address
+@external
+func set_oracle_controller_address{
+    syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+}(oracle_controller_address : felt) -> ():
+    Admin.only_admin()
+    oracle_controller_address_storage.write(oracle_controller_address)
+    return ()
+end
+
+#
+# Helpers
+#
+
+# @dev converts a currency to an asset key
+#      by constructing felt equivalent of "{currency}{denominator}"
+# @param currency: the currency to be turned into an asset_key
+# @param denominator: the value the asset is denominated in including the slash
+# @param denominator_bits: the number of bits needed to represent the denominator
+# @return asset_key: the asset key as felt of "{currency}/{denominator}"
+func _convert_currency_to_asset_key{range_check_ptr}(
+    currency : felt, denominator : felt, denominator_bits : felt
+) -> (asset_key : felt):
+    let (shifted) = _shift_left(currency, 2, denominator_bits)
+    let asset_key = shifted + denominator
```

Comment:
`denominator` needs to include the slash which saves one shift. We could change this.